xkcd.py

```diff
@@ -0,0 +1,45 @@
+import random
+import requests
+
+import discord
+from discord.ext import commands
+
+
+class Xkcd(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name="xkcd")
+    async def cmd_xkcd(self, ctx, number=None):
+        # Daten vom aktuellsten Comic holen, um max zu bestimmen
+        data = requests.get('http://xkcd.com/info.0.json')
```

Comment:
requests ist blocking. Während der HTTP-Request läuft reagiert Boty nicht auf Kommandos etc.
Siehe https://discordpy.readthedocs.io/en/stable/faq.html#what-does-blocking-mean (Beispiel für aoihttp https://github.com/ffaadd/fernuni-bot/blob/module-information/module_information/scrapper.py in der fetch-methode)

Reply:
danke für die Links - du hattest das ja oben schon geschrieben, hatte ich irgendwie überlesen